src/JustSaying/Messaging/MessageSerialization/SystemTextJsonMessageBodySerializer`1.cs

```diff
@@ -0,0 +1,42 @@
+using System.Text.Json;
+using JustSaying.Models;
+
+namespace JustSaying.Messaging.MessageSerialization;
+
+/// <summary>
+/// Provides serialization and deserialization functionality for messages of type <typeparamref name="T"/> using System.Text.Json.
+/// </summary>
+/// <typeparam name="T">The type of message to be serialized or deserialized. Must inherit from <see cref="Message"/>.</typeparam>
+public sealed class SystemTextJsonMessageBodySerializer<T> : IMessageBodySerializer where T: Message
+{
+    private readonly JsonSerializerOptions _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SystemTextJsonMessageBodySerializer{T}"/> class with custom JSON serializer options.
+    /// </summary>
+    /// <param name="options">The custom <see cref="JsonSerializerOptions"/> to use for serialization and deserialization.</param>
+    public SystemTextJsonMessageBodySerializer(JsonSerializerOptions options)
```

Comment:
Is it worth having a default ctor that takes no args and just passes in the DefaultJsonSerializerOptions?